File: percentages.py
class Percentage:
    def __init__(self):
        pass

    def percentage(self, g, pw):
        return (pw/g) * 100

    def percentage_val(self, g, p):
        return (g/10) * (p/10)

    def val_calc(self, p, pw):
        return (pw*100) / p


calc = Percentage()
base = 600
percentage = 30
fraction = 180

print(str(calc.percentage(base, fraction)) + "%")
print(calc.percentage_val(base, percentage))
print(calc.val_calc(percentage, fraction))

File: percentages.c
#include <stdio.h>


int percentage(g, pw) { return (g / pw) * 100; }
int percentage_val(g, p) { return g * p / 100; }
int val_calc(p, pw) { return (pw * 100) / p; }


int main()
{
    int base = 600;
    int percentage = 30;
    int fraction = 180;

    printf("%d\n", percentage(base, fraction));
    printf("%d\n", percentage_val(base, percentage));
    printf("%d\n", val_calc(percentage, fraction));
}
